sdk/messaging/azeventhubs/checkpoints/blob_store.go

```diff
@@ -0,0 +1,390 @@
+			Metadata: blobMetadata,
+		})
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		return uploadResp.LastModified, *uploadResp.ETag, nil
+	}
+}
+
+func nameForCheckpointBlob(a azeventhubs.CheckpointStoreAddress) (string, error) {
+	if a.FullyQualifiedNamespace == "" || a.EventHubName == "" || a.ConsumerGroup == "" || a.PartitionID == "" {
+		return "", errors.New("missing fields for blob name")
+	}
+
+	// checkpoint: fully-qualified-namespace/event-hub-name/consumer-group/checkpoint/partition-id
+	return fmt.Sprintf("%s/%s/%s/checkpoint/%s", a.FullyQualifiedNamespace, a.EventHubName, a.ConsumerGroup, a.PartitionID), nil
+}
+
+func prefixForCheckpointBlobs(a azeventhubs.CheckpointStoreAddress) (string, error) {
+	if a.FullyQualifiedNamespace == "" || a.EventHubName == "" || a.ConsumerGroup == "" {
+		return "", errors.New("missing fields for blob prefix")
+	}
+
+	// checkpoint: fully-qualified-namespace/event-hub-name/consumer-group/checkpoint/
+	return fmt.Sprintf("%s/%s/%s/checkpoint/", a.FullyQualifiedNamespace, a.EventHubName, a.ConsumerGroup), nil
+}
+
+func nameForOwnershipBlob(a azeventhubs.CheckpointStoreAddress) (string, error) {
+	if a.FullyQualifiedNamespace == "" || a.EventHubName == "" || a.ConsumerGroup == "" || a.PartitionID == "" {
+		return "", errors.New("missing fields for blob name")
+	}
+
+	// ownership : fully-qualified-namespace/event-hub-name/consumer-group/ownership/partition-id
+	return fmt.Sprintf("%s/%s/%s/ownership/%s", a.FullyQualifiedNamespace, a.EventHubName, a.ConsumerGroup, a.PartitionID), nil
+}
+
+func prefixForOwnershipBlobs(a azeventhubs.CheckpointStoreAddress) (string, error) {
+	if a.FullyQualifiedNamespace == "" || a.EventHubName == "" || a.ConsumerGroup == "" {
+		return "", errors.New("missing fields for blob prefix")
+	}
+
+	// ownership : fully-qualified-namespace/event-hub-name/consumer-group/ownership/
+	return fmt.Sprintf("%s/%s/%s/ownership/", a.FullyQualifiedNamespace, a.EventHubName, a.ConsumerGroup), nil
+}
+
+func newCheckpointData(metadata map[string]*string) (azeventhubs.CheckpointData, error) {
+	if metadata == nil {
+		return azeventhubs.CheckpointData{}, fmt.Errorf("no checkpoint metadata for blob")
+	}
+
+	sequenceNumberStr, ok := metadata["sequencenumber"]
+
+	if !ok || sequenceNumberStr == nil {
+		return azeventhubs.CheckpointData{}, errors.New("sequencenumber is missing from metadata")
+	}
+
+	sequenceNumber, err := strconv.ParseInt(*sequenceNumberStr, 10, 64)
+
+	if err != nil {
+		return azeventhubs.CheckpointData{}, fmt.Errorf("sequencenumber could not be parsed as an int64: %w", err)
+	}
+
+	offsetStr, ok := metadata["offset"]
+
+	if !ok || offsetStr == nil {
+		return azeventhubs.CheckpointData{}, errors.New("offset is missing from metadata")
+	}
+
+	offset, err := strconv.ParseInt(*offsetStr, 10, 64)
+
+	if err != nil {
+		return azeventhubs.CheckpointData{}, fmt.Errorf("offset could not be parsed as an int64: %w", err)
```

Comment:
I don't think you want `%w` here as that exposes the underlying error type to callers.

Reply:
Generally speaking, I think the default position is that `%w` is the wrong thing to do, unless you really intend to expose the underlying error type.

Reply:
Ah, good point. Just habit when wrapping an error :)

Reply:
Fixed.